pkg/controllers/metrics/nodepool/controller.go

```diff
@@ -94,7 +95,8 @@ func (c *Controller) Name() string {
 
 // Reconcile executes a termination control loop for the resource
 func (c *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
-	ctx = logging.WithLogger(ctx, logging.FromContext(ctx).Named(c.Name()).With("nodepool", req.Name))
+	ctx = log.IntoContext(ctx, log.FromContext(ctx).WithName(c.Name()).WithValues("nodepool", req.Name))
```

Comment:
Is this auto-injected by the logger now? I wonder if we even need to explicitly inject the name of the kind here since I think controller-runtime will just inject the object.Key that is being passed through into the reconcile through the logger by default

Reply:
Do you mean we better drop `WithName(c.Name())` ?